### PR TITLE
OCPBUGS-44786: add support for  the LLC alignment cpumanager policy option

### PR DESCRIPTION
**Pull request by ffromani**
add support for the LLC alignment by pulling ahead of time u/s PR 126750.

### CODE REVIEW

**Comment by kannon92** [comment]
From what I remember, we don't really have a lot of e2e tests on this feature in upstream. How are we going to verify that this works on previous releases?

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 1]
```diff
@@ -138,6 +139,7 @@ func NewStaticPolicy(topology *topology.CPUTopology, numReservedCPUs int, reserv
 	}
 
 	klog.InfoS("Static policy created with configuration", "options", opts)
+	klog.InfoS("Static policy created with configuration", "llcAlignment", llcalign.IsEnabled())
```
Comment (kannon92) [inline pkg/kubelet/cm/cpumanager/policy_static.go:142]: This log jumps out at me a little bit. Maybe we make the log more specific to llcalign?

**Comment by ffromani** [comment]
> From what I remember, we don't really have a lot of e2e tests on this feature in upstream. How are we going to verify that this works on previous releases?

we will have e2e tests in the telco testsuites; and the telco testsuites will run on split-LLC hardware (high end AMD CPUs). 
If we want the e2e tests in openshift/origin, which I'd be happy to had, we will need to figure out how to link machines high end AMD CPUs in openshift CI, because AFAIK there are none.

File: pkg/kubelet/cm/cpumanager/policy_static.go (hunk 1 above)
Comment (ffromani) [reply pkg/kubelet/cm/cpumanager/policy_static.go:142]: yeah, fair point. I'll try something.

File: pkg/kubelet/llcalign/llcalign.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	if _, err := os.Stat(llcAlignmentFilename); err == nil {
+		llcAlignmentEnabled = true
+	}
+}
+
+// TestOnlySetEnabled allows changing the state of management partition enablement
+// This method MUST NOT be used outside of test code
+func TestOnlySetEnabled(enabled bool) {
+	llcAlignmentEnabled = enabled
+}
```
Comment (haircommander) [inline pkg/kubelet/llcalign/llcalign.go:42]: what're the plans for using this?

Comment (ffromani) [reply pkg/kubelet/llcalign/llcalign.go:42]: no real plans yet - too eager copy-paste from `managed` code. I will remove and re-add later if needed.

**Comment by ffromani** [comment]
@haircommander @mrunalp could you PTAL again? the PR implements now the suggested approach I outlined and we agreed upon in https://github.com/openshift/enhancements/pull/1724

**Comment by ffromani** [comment]
/test okd-scos-e2e-aws-ovn

```
{  openshift cluster install failed with cluster bootstrap}
```

File: pkg/kubelet/cm/cpumanager/policy_options.go [hunk 3]
```diff
@@ -56,6 +58,14 @@ func CheckPolicyOptionAvailable(option string) error {
 		return fmt.Errorf("unknown CPU Manager Policy option: %q", option)
 	}
 
+	// must override the base feature gate check. Relevant only for alpha (disabled by default).
+	// for beta options are enabled by default and we totally want to keep the possibility to
+	// disable them explicitly.
+	if alphaOptions.Has(option) && checkPolicyOptionHasEnablementFile(option) {
```
Comment (mrunalp) [inline pkg/kubelet/cm/cpumanager/policy_options.go:64]: wouldn't this put the cluster in TPNU?

Comment (ffromani) [reply pkg/kubelet/cm/cpumanager/policy_options.go:64]: this is the internal check: `alphaOptions` is the internal set of known cpumanager policy options, and this specific guard is meant to bypass the upstream alpha FG check for features which have the enablement file. It was implemented this way (vs extending the existing guard below) to make it as easy as possible to remove later on when the carry is no longer needed. It is meant to be equivalent to:
```
// CheckPolicyOptionAvailable verifies if the given option can be used depending on the Feature Gate Settings.
// returns nil on success, or an error describing the failure on error.
func CheckPolicyOptionAvailable(option string) error {
	if !alphaOptions.Has(option) && !betaOptions.Has(option) && !stableOptions.Has(option) {
		return fmt.Errorf("unknown CPU Manager Policy option: %q", option)
	}

	if alphaOptions.Has(option) {
		if checkPolicyOptionHasEnablementFile(option) {
			return nil
		}
		if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CPUManagerPolicyAlphaOptions) {
			return fmt.Errorf("CPU Manager Policy Alpha-level Options not enabled, but option %q provided", option)
		}
	}

	if betaOptions.Has(option) && !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CPUManagerPolicyBetaOptions) {
		return fmt.Errorf("CPU Manager Policy Beta-level Options not enabled, but option %q provided", option)
	}

	return nil
}
```

**Comment by ffromani** [comment]
/test e2e-agnostic-ovn-cmd

```
{  fail [github.com/openshift/origin/test/extended/apiserver/api_requests.go:406]: Expected
    <[]string | len:1, cap:1>: [
        "Operator \"cluster-samples-operator\" produces more watch requests than expected: watchrequestcount=177, upperbound=118, ratio=1.5",
    ]
to be empty
Ginkgo exit error 1: exit with code 1}
```

**Comment by ffromani** [comment]
```
: [sig-cli][Feature:LegacyCommandTests][Disruptive][Serial] test-cmd: test/cmd/images.sh [apigroup:image.openshift.io] expand_less 	23s
{  fail [github.com/openshift/origin/test/extended/cmd/cmd.go:111]: Expected
    <[]error | len:1, cap:1>: [
        <*errors.errorString | 0xc0012fb510>{
            s: "error waiting for the pod 'test-cmd' to complete:  imagestreamtag tag-c --from-image=quay.io/openshifttest/hello-openshift:openshift' expecting failure and text 'must be of the form <stream_name>:<tag>'\nRunning test/cmd/images.sh:108: executing 'oc create imagestreamtag tag-c:1 -A foo' expecting failure and text 'annotations must be of the form key=value, but is \"foo\"'...\nSUCCESS after 1.000s: test/cmd/images.sh:108: executing 'oc create imagestreamtag tag-c:1 -A foo' expecting failure and text 'annotations must be of the form key=value, but is \"foo\"'\nRunning test/cmd/images.sh:109: executing 'oc create imagestreamtag tag-c:2 --from=mysql --from-image=quay.io/openshifttest/hello-openshift:openshift' expecting failure and text '\\--from and --from-image may not be used together'...\nSUCCESS after 0.000s: test/cmd/images.sh:109: executing 'oc create imagestreamtag tag-c:2 --from=mysql --from-image=quay.io/openshifttest/hello-openshift:openshift' expecting failure and text '\\--from and --from-image may not be used together'\nRunning test/cmd/images.sh:111: executing 'oc get istag/tag:1 -o jsonpath={.image.dockerImageReference}' expecting success and text 'wildfly-centos7.*@sha256:'...\nSUCCESS after 0.000s: test/cmd/images.sh:111: executing 'oc get istag/tag:1 -o jsonpath={.image.dockerImageReference}' expecting success and text 'wildfly-centos7.*@sha256:'\nRunning test/cmd/images.sh:113: executing 'oc get istag/tag-b:1 -o jsonpath={.image.metadata.name}' expecting success and text 'sha256:a2812e358a6495ef37ead2e015b4f680c6d022a99375e095a73de319cd5ad53c'...\nFAILURE after 0.000s: test/cmd/images.sh:113: executing 'oc get istag/tag-b:1 -o jsonpath={.image.metadata.name}' expecting success and text 'sha256:a2812e358a6495ef37ead2e015b4f680c6d022a99375e095a73de319cd5ad53c': the command returned the wrong error code; the output content test failed\nThere was no output from the command.\nStandard error from the command:\nError from server (NotFound): imagestreamtags.image.openshift.io \"tag-b:1\" not found\n[ERROR] hack/lib/cmd.sh:30: `return \"${return_code}\"` exited with status 1.\n",
        },
    ]
to have length 0
Ginkgo exit error 1: exit with code 1}
```

**Pull request merged**